components/availability/src/Availability.svelte

```diff
@@ -246,37 +218,31 @@
     _this = buildInternalProps($$props, manifest, defaultValueMap) as Manifest;
     transformPropertyValues();
 
-    const calendarQuery: CalendarQuery = {
+    // TODO: we probably dont want to expose a list of all a users calendars to the end-user here.
+    const calendarsList = await CalendarStore.getCalendars({
       access_token,
       component_id: id,
       calendarIDs: [], // empty array will fetch all calendars
-    };
-    // TODO: we probably dont want to expose a list of all a users calendars to the end-user here.
-    const calendarsList = await CalendarStore.getCalendars(calendarQuery);
-    loading = false;
+    });
     calendarID = calendarsList?.find((cal) => cal.is_primary)?.id || "";
+    loading = false;
   });
 
   let previousProps = $$props;
   $: {
-    if (JSON.stringify(previousProps) !== JSON.stringify($$props)) {
+    // TODO - Use a library to calculate the diff between props so that
+    // only updated props are reassigned instead of rebuilding the whole object.
```

Comment:
Good option for a tech debt story. We're currently rebuilding all props whenever a passed prop's value changes. This causes all reactive bindings that rely on passed props to fire, even if their prop wasn't the one that was changed. With a library like https://www.npmjs.com/package/deep-diff, we could instead selectively update the internal props with only the diff'd prop.